test/generators/suspenders/install/web_generator_test.rb

```diff
@@ -0,0 +1,87 @@
+require "test_helper"
+require "generators/suspenders/install/web_generator"
+
+module Suspenders
+  module Generators
+    module Install
+      class WebGeneratorTest < Rails::Generators::TestCase
+        include Suspenders::TestHelpers
+
+        tests Suspenders::Generators::Install::WebGenerator
+        destination Rails.root
+        setup :prepare_destination
+        teardown :restore_destination
+
+        test "raises if API only application" do
+          within_api_only_app do
+            assert_raises Suspenders::Generators::APIAppUnsupported::Error do
+              run_generator
+            end
+          end
+        end
+
+        test "invokes generators" do
+          accessibility_generator_mock = mock("accessibility_generator")
+          Suspenders::Generators::AccessibilityGenerator.stubs(:new).returns(accessibility_generator_mock)
+
+          styles_generator_mock = mock("styles_generator")
+          Suspenders::Generators::StylesGenerator.expects(:new).with([], css: "tailwind").returns(styles_generator_mock)
+
+          advisories_generator_mock = mock("advisories_generator")
+          Suspenders::Generators::AdvisoriesGenerator.stubs(:new).returns(advisories_generator_mock)
+
+          inline_svg_generator_mock = mock("inline_svg_generator")
+          Suspenders::Generators::InlineSvgGenerator.stubs(:new).returns(inline_svg_generator_mock)
+
+          facories_generator_mock = mock("facories_generator")
+          Suspenders::Generators::FactoriesGenerator.stubs(:new).returns(facories_generator_mock)
+
+          jobs_generator_mock = mock("jobs_generator")
+          Suspenders::Generators::JobsGenerator.stubs(:new).returns(jobs_generator_mock)
+
+          lint_generator_mock = mock("lint_generator")
+          Suspenders::Generators::LintGenerator.stubs(:new).returns(lint_generator_mock)
+
+          rake_generator_mock = mock("rake_generator")
+          Suspenders::Generators::RakeGenerator.stubs(:new).returns(rake_generator_mock)
+
+          accessibility_generator_mock.expects(:invoke_all).once
+          styles_generator_mock.expects(:invoke_all).once
+          advisories_generator_mock.expects(:invoke_all).once
+          inline_svg_generator_mock.expects(:invoke_all).once
+          facories_generator_mock.expects(:invoke_all).once
+          jobs_generator_mock.expects(:invoke_all).once
+          lint_generator_mock.expects(:invoke_all).once
+          rake_generator_mock.expects(:invoke_all).once
+
+          generator_class.new([], css: "tailwind").invoke_generators
+        end
```

Comment:
I do everything in one test to avoid actually invoking the generators, since each one is stubbed. Otherwise, the `prepare_destination` and `restore_destination` logic would be too much, since it would need to account for all generators.